packages/editor-ui/src/composables/useNodeBase.ts

```diff
@@ -0,0 +1,675 @@
+import { computed, getCurrentInstance, ref } from 'vue';
+
+import type { INodeUi } from '@/Interface';
+import {
+	NO_OP_NODE_TYPE,
+	NODE_CONNECTION_TYPE_ALLOW_MULTIPLE,
+	NODE_INSERT_SPACER_BETWEEN_INPUT_GROUPS,
+	NODE_MIN_INPUT_ITEMS_COUNT,
+} from '@/constants';
+
+import { NodeHelpers, NodeConnectionType } from 'n8n-workflow';
+import type {
+	ConnectionTypes,
+	INodeInputConfiguration,
+	INodeTypeDescription,
+	INodeOutputConfiguration,
+	Workflow,
+} from 'n8n-workflow';
+import { useUIStore } from '@/stores/ui.store';
+import { useWorkflowsStore } from '@/stores/workflows.store';
+import { useNodeTypesStore } from '@/stores/nodeTypes.store';
+import type { BrowserJsPlumbInstance } from '@jsplumb/browser-ui';
+import type { Endpoint, EndpointOptions } from '@jsplumb/core';
+import * as NodeViewUtils from '@/utils/nodeViewUtils';
+import type { EndpointSpec } from '@jsplumb/common';
+import { useDeviceSupport } from 'n8n-design-system';
+import type { N8nEndpointLabelLength } from '@/plugins/jsplumb/N8nPlusEndpointType';
+import { isValidNodeConnectionType } from '@/utils/typeGuards';
+import { useI18n } from '@/composables/useI18n';
+
+export function useNodeBase({
```

Comment:
Would you like to add some tests for this in the same PR? or should we try to do that in a separate PR ?

Reply:
Added some tests for main functions now